app/decorators/service_provider_session_decorator.rb

```diff
@@ -85,11 +76,13 @@ def cancel_link_url
     view_context.new_user_session_url(request_id: sp_session[:request_id])
   end
 
-  def sp_alert(path)
-    path_to_section_map = { new_user_session_path => 'sign_in',
-                            sign_up_email_path => 'sign_up',
-                            new_user_password_path => 'forgot_password' }
-    custom_alert(path_to_section_map[path])
+  def sp_alert(section)
+    return if sp.help_text.nil?
```

Comment:
small suggestion just in case we ever get an empty string
```suggestion
    return if sp.help_text.blank?
```

Reply:
This is carried over from the previous implementation. It's confusingly named, but this value is the _hash_ of texts, not the string itself. We handle blank strings below (`alert.present?`), and there's [a related spec](https://github.com/18F/identity-idp/blob/830445ce48a30311e8d005869e1eed32109361ef/spec/decorators/service_provider_session_decorator_spec.rb#L71-L78) for the specific behavior you're describing.